pkg/policyengine/loadBalancer_test.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2023 The ClusterLink Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policyengine_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	event "github.com/clusterlink-net/clusterlink/pkg/controlplane/eventmanager"
+	"github.com/clusterlink-net/clusterlink/pkg/policyengine"
+)
+
+const (
+	svc1  = "svc1"
+	svc2  = "svc2"
+	svc3  = "svc3"
+	peer1 = "peer1"
+	peer2 = "peer2"
+	peer3 = "peer3"
+)
+
+func addImports(lb *policyengine.LoadBalancer) {
+	// svc1 is imported from peer1 and peer2
+	// svc2 is imported only from peer1
+	// svc3 is imported from peer1 and peer2
+	lb.AddToServiceMap(svc1, peer1)
+	lb.AddToServiceMap(svc1, peer2)
+	lb.AddToServiceMap(svc2, peer1)
+	lb.AddToServiceMap(svc3, peer1)
+	lb.AddToServiceMap(svc3, peer2)
+}
+
+// We repeat lookups enough times to make sure we get all the peers allowed by the relevant policy
+func repeatLookups(t *testing.T, lb *policyengine.LoadBalancer, srcSvc, dstSvc string, peers []string) map[string]bool {
```

Comment:
If it is only for valid peers, maybe you can stop earlier when all array is true

Reply:
Good idea. Will implement in next PR.